### PR TITLE
docs: trimmed rules h1s to just be rule names

### LINKED ISSUE #15471: Documentation formatting change: move descriptions out of h1s
### ESLint version

v8.X

### What problem do you want to solve?

Porting discussion from https://github.com/typescript-eslint/typescript-eslint/issues/4365:

> Right now individual [ESLint] rules' pages have a somewhat unwieldy header based on their description. 

> [TypeScript ESLint] used this style because it's the standard across the ESLint ecosystem.

> In general this is kind of a violation of what a heading is supposed to be, no? I don't know of any precedent for using the top level heading in a file to describe anything like this description. Is this exclusively in ESLint?

### What do you think is the correct solution?

> I'd like to swap that with a header equal to the rule name, and move that description into body text.

With https://eslint.org/docs/rules/array-callback-return as an example:

```diff
- # Enforces return statements in callbacks of array's methods (array-callback-return)
+ # `array-callback-return`
+ 
+ Enforces return statements in callbacks of array's methods.

...
```

### Participation

- [X] I am willing to submit a pull request for this change.

### Additional comments

There are probably accessibility experts who have much more details opinions on whether putting up to a couple dozen words of description text is reasonable for an h1. I suspect not?

Either way, I don't know of other ecosystems that prefer long h1s this way. IMO it hurts readability a bit.

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Moves the description of each rule out of its h1 and into a one-line sentence after.

Fixes #15471.

#### Is there anything you'd like reviewers to focus on?

There were actually quite a few rules that had different text in their docs heading compared to their `meta.docs.description`. ~Separate from this PR I'd suggest adding testing to make sure they stay in sync. I _think_ a good heuristic to fix them all up would be to take the longer between the docs intro paragraph from this PR and the description from the rule itself.~ ~Aha, they're in the makefile! Fixing up now before un-drafting.~ ✅ 

I didn't add a check that the docs page description matches the rule meta docs, because they generally don't. But if you'd like me to make that change in this PR I'd be happy to.

<details>
<summary>Here's a quick script I used to get these changes mostly done...</summary>

```js
// docs/rules/migrate.mjs

import { promises as fs } from "fs";

const mdFiles = (await fs.readdir(".")).filter((x) => x.endsWith(".md"));
const problems = [];

for (const fileName of mdFiles) {
    try {
        await convertFile(fileName);
    } catch (error) {
        problems.push({ error, fileName });
    }
}

if (problems.length) {
    console.error("Problems!");

    for (const { error, fileName } of problems) {
        console.error(`${fileName}: ${error}`);
    }

    process.exitCode = 1;
}

async function convertFile(fileName) {
    const contents = (await fs.readFile(fileName)).toString();
    const [firstLine, ...remainingLines] = contents.split(/\n/);

    const { description, ruleName } = parseMetaFromLine(firstLine);

    const updatedLines = [`# ${ruleName}`, "", description, ...remainingLines];

    await fs.writeFile(fileName, updatedLines.join("\n"));
}

function parseMetaFromLine(line) {
    // Most rules start with the format:
    // # ${Description} (${rule-name})
    const descriptionFirst = line.match(/\# (.+) \((.+)\)/);
    if (descriptionFirst) {
        return {
            description: touchUpDescription(descriptionFirst[1]),
            ruleName: descriptionFirst[2],
        };
    }

    // ...except for a few oddities such as generator-star.md:
    // # ${rule-name}: ${Description}
    const ruleNameFirst = line.match(/# (.+): (.*)/);
    if (ruleNameFirst) {
        return {
            description: touchUpDescription(ruleNameFirst[2]),
            ruleName: ruleNameFirst[1],
        };
    }

    throw new Error("Unknown first line format.");
}

function touchUpDescription(description) {
    description = description[0].toUpperCase() + description.slice(1);

    // In retrospect, I wish I'd done replaces on "allow", "require", etc.
    let [firstWord, ...rest] = description.split(" ");
    if (firstWord.endsWith("e") || firstWord.endsWith("w")) {
        firstWord += "s";
    }
    description = [firstWord, ...rest].join(" ");

    if (!description.endsWith(".")) {
        description += ".";
    }

    return description;
}
```
</details>

### CODE REVIEW

File: docs/rules/eqeqeq.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
-# Require === and !== (eqeqeq)
+# eqeqeq
+
+Requires the use of === and !==.
```
Comment (snitin315) [inline docs/rules/eqeqeq.md:3]: This one caught my eye. 

```suggestion
Requires the use of `===` and `!==`.
```

Comment (snitin315) [reply docs/rules/eqeqeq.md:3]: This seems inconsistent throughout the docs, sometimes technical keywords are highlighted with code sometimes not.

**Review by aladdin-add — APPROVED** [review]
LGTM~

File: docs/rules/eqeqeq.md (hunk 1 above)
Comment (JoshuaKGoldberg) [reply docs/rules/eqeqeq.md:3]: Heh, yeah. I'll take a pass through the rules and fix up what I can find.

**Review by snitin315 — APPROVED** [review]
LGTM. Thank you for contributing ❤️

**Pull request merged**